libnd4j/include/ops/declarable/generic/nlp/cbow.cpp

```diff
@@ -89,30 +89,42 @@ CONFIGURABLE_OP_IMPL(cbow_inference, 6, 6, true, -2, -2) {
   const std::vector<sd::LongType> *contextShape = contextSize;
 
   std::vector<sd::LongType> *lockedWordsSize = new std::vector<sd::LongType>();
-  contextSize->push_back(lockedWords->size());
+  lockedWordsSize->push_back(lockedWords->size());
   const std::vector<sd::LongType> *lockedWordsShape = lockedWordsSize;
 
+  //sd_printf("Before converting args to ndarrays\n",0);
 
-  auto indicesArrOne = NDArrayFactory::create('c',*indicesShape,*indicesVec);
+  auto indicesArrOne = indicesVec->size() > 0 ? NDArrayFactory::create('c',*indicesShape,*indicesVec) : NDArrayFactory::empty<sd::LongType>();
   auto indicesArr = new NDArray(indicesArrOne);
+  //sd_printf("After converting args to ndarrays indices\n",0);
 
-  auto codesArrOne = NDArrayFactory::create('c',*codesShape,*codesVec);
+  auto codesArrOne = codesVec->size() > 0 ?  NDArrayFactory::create('c',*codesShape,*codesVec) :  NDArrayFactory::empty<sd::LongType>();
   auto codesArr = new NDArray(codesArrOne);
+  //sd_printf("After converting args to ndarrays codes\n",0);
 
-  auto contextArrOne = NDArrayFactory::create('c',*contextShape,*contextVec);
+  //sd_printf("Context arr size %d Context shape entry is %d\n",context->size(),contextShape->at(0));
+  //sd_printf("Context: elements: ",0);
+  for(int i = 0; i < context->size(); i++) {
+    //sd_printf(" %d ",context->at(i));
+  }
+  //sd_printf("\n",0);
```

Comment:
Let's not have commented out code like that. And in particular not have a for loop doing nothing.

Reply:
Fixed